returnn/tf/layers/rec.py

```diff
@@ -7771,11 +7774,13 @@ def transform_config_dict(cls, d, network, get_layer):
     else:
       d["masked_from"] = None
     super(MaskedComputationLayer, cls).transform_config_dict(d, network=network, get_layer=get_layer)
+    mask = get_layer(d["mask"])
```

Comment:
See also the logic below for `mask`.

Reply:
Ah okay. I need the mask information in the `_create_template` function. Should I just move the `d["mask"]` logic in front of the `_create_template` call?

Reply:
> Should I just move the `d["mask"]` logic in front of the `_create_template` call?

This would make the test case work again. I just checked on my local computer

Reply:
I'm not really sure how you mean exactly. And also, if you still have the logic for dependency resolution (it should not check `mask` when not needed), `masked_from` and `parent_layer_cache` correct then.

Reply:
I mean whether I can just put 
```
if masked_from and not parent_layer_cache:
  # We explicitly do not want to have these as deps.
  d["mask"] = None
else:
  d["mask"] = get_layer(d["mask"])
```
before `d["_layer_class"], d["_layer_desc"] = cls._create_template(...)`

and then remove `mask = get_layer(d["mask"])`